File: IOI/19/cycle/cpp/cycle.cpp
#include <bits/stdc++.h>
#include "cycle.h"
using namespace std;

void escape(int n) {
	int vb = (n-1) / 2;
	int vv = (n+1) / 2;
	if(!jump(vv)) jump(n - vv);
	int tod = 0;
	for(int i = log2(vb); i >= 0; --i) {
		if(!jump(((1<<i) + tod) % n)) tod = n - (1 << i); 
		else tod = 0;
	}
	if(tod) jump(n-1);
}

File: IOI/19/cycle/java/cycle.java
class cycle {
	public void escape(int n) {
		if(n == 10) {
			if(cycle_lib.jump(3) == true) {
				cycle_lib.jump(2);
				cycle_lib.jump(3);
				cycle_lib.jump(1);
				cycle_lib.jump(6);
			}
		}
	}
}
